botocore/auth.py

```diff
@@ -367,7 +368,15 @@ def _modify_request_before_signing(self, request):
         super(S3SigV4Auth, self)._modify_request_before_signing(request)
         if 'X-Amz-Content-SHA256' in request.headers:
             del request.headers['X-Amz-Content-SHA256']
-        request.headers['X-Amz-Content-SHA256'] = self.payload(request)
+
+        # S3 allows optional body signing, so to minimize the performance
+        # impact, we opt to not SHA256 sign the body. Instead, we MD5 it.
+        # We will fall back to SHA256 when MD5 is not available, such as with
+        # FIPS systems.
+        if MD5_AVAILABLE:
```

Comment:
Can't we just check if there's a Content-MD5 value in the request headers?  Seems more definitive.